src/internal/coredb/projects.go

```diff
@@ -0,0 +1,171 @@
+	ID          ID        `db:"id"`
+	Name        string    `db:"name"`
+	Description string    `db:"description"`
+	CreatedAt   time.Time `db:"created_at"`
+	UpdatedAt   time.Time `db:"updated_at"`
+}
+
+// Next advances the iterator by one row. It returns a stream.EOS when there are no more entries.
+func (iter *ProjectIterator) Next(ctx context.Context, dst **pfs.ProjectInfo) error {
+	if dst == nil {
+		return errors.Wrap(fmt.Errorf("project is nil"), "failed to get next project")
+	}
+	var err error
+	if iter.index >= len(iter.projects) {
+		iter.index = 0
+		iter.offset += iter.limit
+		iter.projects, err = listProject(ctx, iter.tx, iter.limit, iter.offset)
+		if err != nil {
+			return errors.Wrap(err, "failed to list project page")
+		}
+		if len(iter.projects) == 0 {
+			return stream.EOS()
+		}
+	}
+	row := iter.projects[iter.index]
+	projectTimestamp, err := types.TimestampProto(row.CreatedAt)
+	if err != nil {
+		return errors.Wrap(err, "failed to convert time.Time to proto timestamp")
+	}
+	*dst = &pfs.ProjectInfo{
+		Project:     &pfs.Project{Name: row.Name},
+		Description: row.Description,
+		CreatedAt:   projectTimestamp}
+	iter.index++
+	return nil
+}
+
+// ListProject returns a ProjectIterator that exposes a Next() function for retrieving *pfs.ProjectInfo references.
+func ListProject(ctx context.Context, tx *pachsql.Tx) (*ProjectIterator, error) {
+	limit := 100
+	page, err := listProject(ctx, tx, limit, 0)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list projects")
+	}
+	iter := &ProjectIterator{
+		projects: page,
+		limit:    limit,
+		tx:       tx,
+	}
+	return iter, nil
+}
+
+func listProject(ctx context.Context, tx *pachsql.Tx, limit, offset int) ([]projectRow, error) {
+	var page []projectRow
+	if err := tx.SelectContext(ctx, &page, "SELECT name,description,created_at FROM core.projects ORDER BY id ASC LIMIT $1 OFFSET $2", limit, offset); err != nil {
+		return nil, errors.Wrap(err, "could not get project page")
+
+	}
+	return page, nil
+}
+
+// CreateProject creates an entry in the core.projects table.
+func CreateProject(ctx context.Context, tx *pachsql.Tx, project *pfs.ProjectInfo) error {
+	_, err := tx.ExecContext(ctx, "INSERT INTO core.projects (name, description) VALUES ($1, $2);", project.Project.Name, project.Description)
+	//todo: insert project.authInfo into auth table.
+	return errors.Wrap(err, "failed to create project")
+}
+
+// DeleteProject deletes an entry in the core.projects table.
+func DeleteProject(ctx context.Context, tx *pachsql.Tx, projectName string) error {
+	result, err := tx.ExecContext(ctx, "DELETE FROM core.projects WHERE name = $1;", projectName)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete project")
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return errors.Wrap(err, "could not get affected rows")
+	}
+	if rowsAffected == 0 {
+		return errors.Wrap(fmt.Errorf("project %s does not exist", projectName), "failed to delete project")
```

Comment:
You may wish to return some sort of structured error, or just a wrapped `ErrNotExist`/`ErrProjectNotExist`/something.